spec/shared_examples/checkout.rb

```diff
@@ -0,0 +1,343 @@
+      expect(page).to have_css('.completed', text: 'Address')
+      expect(page).to have_css('.active', text: 'Confirm')
+      expect(page).to have_css('.last', text: 'Complete')
+
+      expect(page).to have_css('.panel-heading', text: 'Confirm')
+      expect(page).to have_css('.col-xs-6', text: 'Billing Address')
+      expect(page).to have_css('.col-xs-6', text: 'Shipping Address')
+      expect(page).to have_css('.col-xs-6', text: 'Shipments')
+      expect(page).to have_css('.col-xs-6', text: 'Payment Information')
+
+      expect(page).to have_css('.active', text: 'Item')
+      expect(page).to have_css('.active', text: 'Price')
+      expect(page).to have_css('.active', text: 'Qty')
+      expect(page).to have_css('.active', text: 'Total')
+      expect(page).to have_css('#line-items', text: 'Ruby on Rails Bag')
+      expect(page).to have_css('#line-items', text: 'Ruby on Rails Tote')
+
+      expect(page).to have_css('.total', text: 'Subtotal')
+      expect(page).to have_css('.total', text: 'Shipping')
+      expect(page).to have_css('.total', text: 'Tax')
+      expect(page).to have_css('.total', text: 'Order Total')
+
+      expect(page).to have_button('Place Order')
+    end
+  end
+
+  it 'can place the order' do
+    click_button 'Place Order'
+
+    aggregate_failures do
+      expect(page).to have_css('.alert-notice', text: 'Your order has been processed successfully')
+      expect(page).to have_css('.button', text: 'Back to Store')
+    end
+  end
+end
+
+shared_examples 'can edit on Confirmation step' do
+  before do
+    save_address(address)
+    save_delivery('UPS Two Day (USD)')
+    save_payment(credit_card)
+  end
+
+  context 'using Navigation' do
+    it 'Billing Address' do
+      find('.completed', text: 'Address').click
+      expect(page).to have_css('.active', text: 'Address')
+
+      save_address(address2)
+      save_delivery('UPS Two Day (USD)')
+      save_payment(credit_card)
+
+      order_bill_address = find('div[data-hook="order-bill-address"]')
+
+      within(order_bill_address) do
+        address2.to_hash.each do |key, value|
+          value = Madison.get_abbrev(value) if key == :state
+
+          expect(order_bill_address).to have_content(value)
+        end
+      end
+    end
+
+    it 'Shipping method' do
+      find('.completed', text: 'Delivery').click
+      expect(page).to have_css('.active', text: 'Delivery')
+
+      save_delivery('UPS One Day (USD)')
+      save_payment(credit_card)
+
+      expect(page).to have_css('.delivery', text: 'From default via UPS One Day (USD)')
+    end
+    
+    it 'Payment' do
+      find('.completed', text: 'Payment').click
+      expect(page).to have_css('.active', text: 'Payment')
+
+      save_payment(credit_card2)
+
+      expect(page).to have_css('.payment-info', text: 'Ending in 2222')
+    end
+  end
+
+  context 'using button' do
+    it 'Billing Address' do
+      find('.steps-data').first('h4 a').click
```

Comment:
I would use here `find('h4', text: 'Billing Address').click_link 'Edit'`, it's more readable